src/ConnectedMode/UI/ManageConnections/ManageConnectionsViewModel.cs

```diff
@@ -32,16 +33,28 @@ public class ManageConnectionsViewModel(IConnectedModeServices connectedModeServ
 
         public async Task LoadConnectionsWithProgressAsync()
         {
-            var validationParams = new TaskToPerformParams<AdapterResponse>(LoadConnectionsAsync, UiResources.LoadingConnectionsText, UiResources.LoadingConnectionsFailedText);
+            var validationParams = new TaskToPerformParams<AdapterResponse>(
+                async () => await SafeExecuteActionAsync(InitializeConnections),
+                UiResources.LoadingConnectionsText,
+                UiResources.LoadingConnectionsFailedText);
             await ProgressReporterViewModel.ExecuteTaskWithProgressAsync(validationParams);
         }
 
-        internal async Task<AdapterResponse> LoadConnectionsAsync()
+        public async Task CreateConnectionsWithProgressAsync(Connection connection, ICredentialsModel credentialsModel)
```

Comment:
It is called by the xaml.cs, that't why I made it public. And it should be accessible to anybody outside, because this is the entry point to really create a connection (and not all the other internal methods, which are just helper methods)

Reply:
`internal` is for assembly scope and the xaml is in the same assembly. Correct me if I'm wrong but internal should work in this case. Similar to the `internal void RemoveConnection`.

It seems that those can be `internal`
* `LoadConnectionsWithProgressAsync`
* `CreateConnectionsWithProgressAsync`
* `RemoveConnection`

and those `private`
* `SafeExecuteActionAsync`
* `InitializeConnections`
* `CreateNewConnection`
* `AddConnectionViewModel`

Reply:
From a technical point of view, you are right. Everything can be made internal. I was just talking from a principle point of view: the public methods are the entry points that are exposed and that viewmodel offers to the exterior (not only to the .xaml.cs, but to any class that might have a reference to it)

I can make it internal, sure. I have no strong opinion about it. 

But the methods that you pointed out, I won't make them private for the following reasons:
- they are called directly by the test assemble. To be reachable they must be at least internal
- to make them private, I would really need to re-write a lot of tests and the question is: is the time invested in that worth the effort? As we are already kind of behind the schedule

Reply:
As discussed on the call, this should not change here, it's a lot of effort and not in the context of this PR.
To prevent such cases, for future reference:
1. such changes should be handled on separate PRs
2. changes should be consistent
3. keep the method accessibility as low as necessary
4. access modifiers should not be driven by the tests